google-cloud-pub-sub/src/test/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSpec.scala

```diff
@@ -67,6 +68,39 @@ class GooglePubSubSpec extends FlatSpec with MockitoSugar with ScalaFutures with
     result.futureValue shouldBe Seq(Seq("id1"))
   }
 
+  ignore should "publish the message without auth when emulated" in {
+    val mockHttpApi = mock[HttpApi]
+    val googlePubSub = new GooglePubSub {
+      val httpApi = mockHttpApi
+
+      def getSession(clientEmail: String, privateKey: PrivateKey) = ???
+    }
+    val flow = googlePubSub.publish(
+      projectId = TestCredentials.projectId,
+      apiKey = TestCredentials.apiKey,
+      clientEmail = TestCredentials.clientEmail,
+      privateKey = TestCredentials.privateKey,
+      topic = "topic2"
+    )
+
+    val request = PublishRequest(Seq(PubSubMessage(messageId = "2", data = base64String("Hello Google!"))))
+
+    val source = Source(List(request))
+
+    // FIXME: not working, returns false
+    when(mockHttpApi.isEmulated).thenReturn(true)
```

Comment:
I would need help here, cannot force this mock to return `true`

Reply:
Looks to me like `Mockito` is not the best choice for Scala projects
https://github.com/mockito/mockito/issues/1102

Reply:
You can create a manual mock instance with 
```
    val mockHttpApi = new HttpApi {
      val PubSubGoogleApisHost: String = "..."
      val GoogleApisHost: String = "..."

      override def isEmulated: Boolean = true

      override def publish(
          project: String,
          topic: String,
          maybeAccessToken: Option[String],
          apiKey: String,
          request: PublishRequest
      )(implicit as: ActorSystem, materializer: Materializer): Future[Seq[String]] =
        Future.successful(Seq("id2"))
    }
```

If everything else in shape in this PR?

Reply:
I can say no, just because pulling from emulator also hits to my comment stated below client tries to authenticate against emulator when pulling, acknowledging etc. So long story short all operations should be in `maybeAccessToken`.